src/octree/batch_iterator.rs

```diff
@@ -151,16 +148,27 @@ impl<'a> BatchIterator<'a> {
     where
         F: FnMut(PointData) -> Result<()>,
     {
-        let mut point_stream =
-            PointStream::new(self.batch_size, self.local_from_global.clone(), &mut func);
-        let mut iterator: Box<Iterator<Item = Point>> = match &self.culling {
-            PointCulling::Any() => Box::new(self.octree.all_points()),
-            PointCulling::Aabb(aabb) => Box::new(self.octree.points_in_box(aabb)),
-            PointCulling::Obb(obb) => Box::new(self.octree.points_in_obb(obb)),
-            PointCulling::Frustum(frustum) => Box::new(self.octree.points_in_frustum(frustum)),
-            PointCulling::OrientedBeam(beam) => Box::new(self.octree.points_in_oriented_beam(beam)),
-        };
-        iterator.try_for_each(|point: Point| point_stream.push_point_and_callback(point))?;
-        point_stream.callback()
+        //todo (catevita) mutable function parallelization
+        let local_from_global = self
+            .point_location
+            .global_from_local
+            .clone()
+            .map(|t| t.inverse());
+        let mut point_stream = PointStream::new(self.batch_size, local_from_global, &mut func);
+        // nodes iterator: retrieve nodes
+        let node_id_iterator = self.octree.nodes_in_location(self.point_location);
+        // operate on nodes
+        for node_id in node_id_iterator {
+            let point_iterator = self.octree.points_in_node(self.point_location, node_id);
+            for point in point_iterator {
+                if let Err(err) = point_stream.push_point_and_callback(point) {
```

Comment:
Can we just do `point_stream.push_point_and_callback(point)?;` instead?

Reply:
done!